src/videojs-contrib-hls.js

```diff
@@ -139,13 +139,37 @@ export default class HlsHandler extends Component {
         this.options_[option] = this.source_[option];
       }
     });
-    this.masterPlaylistController_ = new MasterPlaylistController({
-      url: this.source_.src,
-      withCredentials: this.options_.withCredentials,
-      currentTimeFunc: this.tech_.currentTime.bind(this.tech_),
-      mediaSourceMode: this.options_.mode,
-      hlsHandler: this,
-      externHls: Hls
+    this.options_.url = this.source_.src;
+    this.options_.tech = this.tech_;
+    this.options_.externHls = Hls;
+    this.options_.bandwidth = this.bandwidth;
+    this.masterPlaylistController_ = new MasterPlaylistController(this.options_);
+
+    // re-expose some internal objects for backwards compatibility with < v2
+    this.playlists = this.masterPlaylistController_.masterPlaylistLoader_;
+    this.mediaSource = this.masterPlaylistController_.mediaSource;
+
+    // Proxy assignment of some properties to the master playlist
+    // controller. Using a custom property for backwards compatibility
+    // with < v2
+    Object.defineProperties(this, {
+      selectPlaylist: {
+        get() {
+          return this.masterPlaylistController_.selectPlaylist
+            .bind(this.masterPlaylistController_);
+        },
+        set(selectPlaylist) {
+          this.masterPlaylistController_.selectPlaylist = selectPlaylist;
```

Comment:
Any function passed in to act as `selectPlaylist` expects `this` to be equal to the HlsHandler object and have access to it's non-underlined functions specifically.